drag_and_drop_v2/public/js/drag_and_drop_edit.js

```diff
@@ -183,12 +183,22 @@ function DragAndDropEditBlock(runtime, element) {
                         remove: function(e) {
                             var $el = $(e.currentTarget).closest('.zone-row'),
                                 classes = $el.attr('class'),
-                                id = classes.slice(classes.indexOf('zone-row') + 9);
+                                id = classes.slice(classes.indexOf('zone-row') + 9),
+                                // The id is of the form "zone-<index>".  We extract the index by
+                                // slicing off the first five letters and "casting" to integer.
+                                index = +id.slice(5),
+                                i;
```

Comment:
@smarnach @bradenmacdonald also, is I'm the only one that find declaring a loop variable at the top of the function a strange practice? I.e. instead of `for (var i=...)` do `var i; for (i=...`?